### PR TITLE
Made failed uploads return a HTTP 400 response (#792)

### PR DESCRIPTION
**Pull request by KyleFung**
Failed uploads through Ajax interface reported HTTP code 200 when it
should be 400. There are other similar errors in this module, but this
commit only fixes failed uploads.

### CODE REVIEW

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java [hunk 1]
```diff
@@ -1749,7 +1754,7 @@ private void ajaxHandleUpload(HttpServletRequest req,
           error =
               error.substring(0, 512) + "<br>Too many errors to display.<br>";
         }
-        ret.put("error", "Installation Failed.<br>" + error);
```
Comment (HappyRay) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java]: Is it necessarily a client error?

Comment (KyleFung) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java]: It is impossible from here to tell if it is 400 or 500. I suppose we should err on the side of caution and return 500.

**Comment by HappyRay** [comment]
As discussed, pls create a ticket to track future enhancement including our discussion of the alternative approach of using an exception to return the errors.

**Comment by KyleFung** [comment]
Yeah I made it here: #795. I forgot to reference it in this pull request.

**Pull request merged**